Update DiskFileItem.java: Avoiding NPE when not explicitely initializ…

OleHornischer opened the pull request:
…ing the outputstream

When accessing e.g. getSize() on a DiskFileItem before having called getOutputStream() a NullPointerException would be thrown. Consequent use of the "intelligent" getter already in place avoids this issue.

garydgregory commented:
Hi @OleHornischer,

1) Can you update your patch based on our Git master.
2) Can you provide a unit test that shows the bug in action before if your patch is not applied.

Gary

OleHornischer commented:
Hi @garydgregory 
I created a Unit test illustrating the issue. However, since this is my first pull request, I am not sure what you mean by updating the patch. I forked the trunk and changed only the files required for my change. 
Do I rather need to create a branch in your repository instead?

Ole

garydgregory commented:
You might have to catch up your fork to the HEAD of the master branch. See https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/

OleHornischer commented:
Thanks for the tip. I updated my repository in accordance to the script on the linked blog post.
Hope things are in order now.

garydgregory commented:
Don't you still get an NPE if you call DiskFileItem.isInMemory()?

You should really have one @Test method for each DiskFileItem method you modified to show where NPEs have been fixed.

OleHornischer commented:
Hi @garydgregory,
yes, that potential NullPointerException I also addressed in my fix, as well as others. 
I updated the sources now to handle exceptions and fixed a few compile errors. The test now includes a test for all methods that have been fixed.

Ole
